### DISCUSSION TITLE
Delete recycle bin files after x Days

### DISCUSSION THREAD

**Discussion by bremmercandia** [drakkan/sftpgo#1701] [Q&A]
Does anyone know how to configure a Event Rule to trigger a action who deletes files older than X days?

**Comment by nzx-jsantaana** [comment]
This would be a case for the Data Retention action, run from a Schedule trigger rule.  CAUTION: I haven't actually tested this myself, and Data Retention can delete your files (that's kind of the point), so make backups first.

The docs are not as clear as they might be, but from [this part of the Event Manager page](https://sftpgo.github.io/2.6/eventmanager/#:~:text=data%20retention%20check%2C,trigger%20is%20a%20schedule), it _looks_ like the action will trigger once for each user matching the trigger.  Actions with built-in paths use the context of the user that triggered the action, so if all of your users have individual recycle directories that they see as "/recycle-bin" when logged in, you could just have one Action:
* Type: Data retention check
* one entry under "Data retention":
  * Path: /recycle-bin/
  * Hours: 24*X (eg, 72 to delete files after three days)

Then you'll need the Rule to trigger it:
* Trigger: Schedules
* one entry under "Schedules": eg, to run daily at 7:00 PM UTC
  * Hour: 19
  * Day of week: *
  * Day of month: *
  * Month: *
* one entry under "Name/Group name/Role name filters": eg, to apply to all users
  * Name: *
* Actions: the Data Retention action you previously defined

If you have different recycle-bin arrangements for different users, you will need to either create multiple "Data retention check" actions (and trigger them selectively so the right users run the right actions), or trigger the action as a user who can see the whole directory tree and set up separate path entries in the Action definition (be sure that the trigger only fires for one user in this case).

REMINDER: _I haven't tested this._  Back up your files, use VM snapshots if you can, and try it in a test environment.  Don't blame me if something goes wrong and all your files get eaten, because I won't be able to help.
